code/game/objects/items/portable_vendor.dm

```diff
@@ -211,3 +211,75 @@
 							list("De Void of Soul", 51, /obj/item/clothing/under/liaison_suit/galaxy_blue, "white", "The latest in ultrafashion. for those with a cool demeanor."),
 							list("Pulsar Gonne", 51, /obj/item/clothing/under/liaison_suit/galaxy_red, "white", "The latest in ultrafashion. for those with a fiery temper.")
 							)
+
+// A marine version of the portable vendor, points are not regenerated.
+/obj/item/portable_vendor/marine
```

Comment:
i'm not a fan of how this has been done.
having a physical item for this feels pretty wrong gameplay wise, especially given it doesnt stop you keeping it as a pocket resupply, doesnt self delete when empty and isnt locked to the marine it was given to.

A verb might be better imo.

Reply:
I can do the latter two though. Though I don't really understand the first one, like... you can simply keep this item to quickly resupply yourself? It literally has one point of usage and points cannot be regenerated.

`/obj/item/portable_vendor/marine/process()
	points = min(max_points, points+0)`

Reply:
You can scavenge a gun somewhere and keep this in your pocket for a free resupply